runtime/interpreter/interpreter.go

```diff
@@ -3627,33 +3635,29 @@ func (interpreter *Interpreter) authAccountLinkFunction(addressValue AddressValu
 }
 
 func (interpreter *Interpreter) accountGetLinkTargetFunction(addressValue AddressValue) *HostFunctionValue {
+
+	address := addressValue.ToAddress()
+
 	return NewHostFunctionValue(
 		func(invocation Invocation) Value {
 
-			address := addressValue.ToAddress()
-
 			capabilityPath := invocation.Arguments[0].(PathValue)
 
-			capabilityKey := PathToStorageKey(capabilityPath)
-
-			value := interpreter.ReadStored(address, capabilityKey)
-
-			switch value := value.(type) {
-			case NilValue:
-				return value
```

Comment:
Is it no longer possible to get a `NilValue`  from `interpreter.ReadStored()`?

Reply:
I refactored the internal storage API in 97c952d81aa097191bdf0e0335d2dd74f344b99a: 
Previously, reads returned an optional, `NilValue` if the read returns nothing, and `SomeValue` if there is a returned value. No the function returns a Go `nil` if the read returns nothing, and the stored value itself if there is one, without wrapping it in a `SomeValue`. (Similarly I refactored `writeStored` from `OptionalValue` to Go `nil`/`Value`). This removes the rather unnecessary boxing / allocations, as suggested by @fxamacker 👍